compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/vectorization.mlir

```diff
@@ -0,0 +1,70 @@
+// RUN: iree-opt  --iree-amdaie-vectorization %s | FileCheck %s
+
+#map = affine_map<(d0, d1, d2) -> (d0, d2)>
+#map1 = affine_map<(d0, d1, d2) -> (d2, d1)>
+#map2 = affine_map<(d0, d1, d2) -> (d0, d1)>
+module {
+
+// Test that a matmul (as a generic operation) with bf16 operands
+// is vectorized to a vector.contract operation.
+// CHECK-LABEL: func @matmul_m4_n4_k8_bf16_f32
+func.func @matmul_m4_n4_k8_bf16_f32(%arg0: tensor<4x8xbf16>, %arg1: tensor<8x4xbf16>, %arg2: tensor<4x4xf32>) -> tensor<4x4xf32> {
+  // CHECK-DAG: %[[cst:.*]] = arith.constant 0.000000e+00 : f32
+  // CHECK-DAG: %[[c0:.*]] = arith.constant 0 : index
+  // CHECK-DAG: %[[cst_0:.*]] = arith.constant 0.000000e+00 : bf16
+  // CHECK-DAG: vector.transfer_read{{.*}} tensor<4x8xbf16>, vector<4x8xbf16>
+  // CHECK-DAG: vector.transfer_read{{.*}} tensor<8x4xbf16>, vector<8x4xbf16>
+  // CHECK-DAG: vector.transfer_read{{.*}} tensor<4x4xf32>, vector<4x4xf32>
+  // CHECK: vector.contract{{.*}}vector<4x8xf32>, vector<8x4xf32> into vector<4x4xf32>
```

Comment:
shouldnt this be `vector.contract{{.*}}vector<4x8xbf16>, vector<8x4xbf16> into vector<4x4xf32>` ?

Reply:
There is some casting I've omitted from CHECK here (but will add in for clarity), looks something like:

```
%58 = arith.extf %49 : vector<1x1x1x1x4x8xbf16> to vector<1x1x1x1x4x8xf32>
%59 = arith.extf %53 : vector<1x1x1x1x8x4xbf16> to vector<1x1x1x1x8x4xf32>
```

In the pass `LowerVectorToAIEVec` (run in aie2xclbin) this contract becomes

```
%71 = aievec.matmul %69, %70, %68 : vector<4x8xbf16>, vector<8x4xbf16> into vector<4x4xf32>
```  

It is a bit confusing and not the most direct path.